Core/include/Acts/TrackFitting/BetheHeitlerApprox.hpp

```diff
@@ -0,0 +1,322 @@
+  constexpr static double lowerLimit = 0.10;
+  constexpr static double higherLimit = 0.20;
+
+ private:
+  Data m_low_data;
+  Data m_high_data;
+  bool m_low_transform;
+  bool m_high_transform;
+
+ public:
+  /// Construct the Bethe-Heitler approximation description
+  ///
+  /// @param low_data data for the lower x/x0 range
+  /// @param high_data data for the higher x/x0 range
+  /// @param transform wether the data need to be transformed (see Atlas code)
+  constexpr AtlasBetheHeitlerApprox(const Data &low_data, const Data &high_data,
+                                    bool low_transform, bool high_transform)
+      : m_low_data(low_data),
+        m_high_data(high_data),
+        m_low_transform(low_transform),
+        m_high_transform(high_transform) {}
+
+  /// Returns the number of components the returned mixture will have
+  constexpr auto numComponents() const { return NComponents; }
+
+  /// Checks if an input is valid for the parameterization
+  ///
+  /// @param x input in terms of x/x0
+  constexpr bool validXOverX0(ActsScalar x) const { return x < higherLimit; }
+
+  /// Generates the mixture from the polynomials and reweights them, so
+  /// that the sum of all weights is 1
+  ///
+  /// @param x The input in terms of x/x0 (pathlength in terms of radiation length)
+  auto mixture(ActsScalar x) const {
+    // Build a polynom
+    auto poly = [](ActsScalar xx,
+                   const std::array<ActsScalar, PolyDegree + 1> &coeffs) {
+      ActsScalar sum{0.};
+      for (const auto c : coeffs) {
+        sum = xx * sum + c;
+      }
+      throw_assert(std::isfinite(sum), "polynom result not finite");
+      return sum;
+    };
+
+    // Lambda which builds the components
+    auto make_mixture = [&](const Data &data, double xx, bool transform) {
+      // Value initialization should garanuee that all is initialized to zero
+      std::array<detail::GaussianComponent, NComponents> ret{};
+      ActsScalar weight_sum = 0;
+      for (int i = 0; i < NComponents; ++i) {
+        // These transformations must be applied to the data according to ATHENA
+        // (TrkGaussianSumFilter/src/GsfCombinedMaterialEffects.cxx:79)
+        if (transform) {
+          ret[i] = detail::inverseTransformComponent(
+              poly(xx, data[i].weightCoeffs), poly(xx, data[i].meanCoeffs),
+              poly(xx, data[i].varCoeffs));
+        } else {
+          ret[i].weight = poly(xx, data[i].weightCoeffs);
+          ret[i].mean = poly(xx, data[i].meanCoeffs);
+          ret[i].var = poly(xx, data[i].varCoeffs);
+        }
+
+        weight_sum += ret[i].weight;
+      }
+
+      for (int i = 0; i < NComponents; ++i) {
+        ret[i].weight /= weight_sum;
+      }
+
+      return ret;
+    };
+
+    // Return no change
+    if (x < noChangeLimit) {
+      std::array<detail::GaussianComponent, NComponents> ret{};
+
+      ret[0].weight = 1.0;
+      ret[0].mean = 1.0;  // p_initial = p_final
+      ret[0].var = 0.0;
+
+      return ret;
```

Comment:
what happens to the other components?

Reply:
Good point, at the moment they are ignored, but I always wanted to change to `boost::static_vector`. That would be more clear